### ISSUE TITLE
Unclear how a primary key is detected

### ISSUE THREAD

**Issue by LukasMa** [go-pg/pg#257]
I think it is unclear how a primary key is detected in a model definition. I read about three methodes in the quickstart section:
1. Primary Key is first attribute in model struct
2. Attribute 'Id' is automatically detected as PK
3. An attribute can be annotated as PK by `sql:",pk"`

However, I build the following model:
```go
// User model
type User struct {
	TgIdent int64 `sql:",pk"`
	ID      int
	DoPat   string
}
```
And I still can't use `db.Delete(user)` to delete a record because the PK is `ID` even if it's not the first attribute and annotated.

So:
Shouldn't the annotation have the strongest impact on the selection of the PK and not the attribute name?

**Comment by vmihailenco** [comment]
>Shouldn't the annotation have the strongest impact on the selection of the PK and not the attribute name?

It should. This is an overlook.

**Comment by vmihailenco** [comment]
Should be fixed in v4.6.3
